test/drivers/test_suites/syscall_exit_suite/process_vm_writev_x.cpp

```diff
@@ -0,0 +1,139 @@
+#include "../../event_class/event_class.h"
+#include <gtest/gtest.h>
+
+#ifdef __NR_process_vm_writev
+
+TEST(SyscallExit, process_vm_writevX_failure)
+{
+	auto evt_test = get_syscall_event_test(__NR_process_vm_writev, EXIT_EVENT);
+
+	evt_test->enable_capture();
+
+	/*=============================== TRIGGER SYSCALL ===========================*/
+
+	size_t res = syscall(__NR_process_vm_writev, getpid(), (void*)(0x41414141), 0, (void*)(0x42424242), 0, 0);
+	assert_syscall_state(SYSCALL_FAILURE, "process_vm_writev", res, EQUAL, 0);
+
+	/*=============================== TRIGGER SYSCALL ===========================*/
+
+	evt_test->disable_capture();
+
+	evt_test->assert_event_presence();
+
+	if(HasFatalFailure())
+	{
+		return;
+	}
+
+	evt_test->parse_event();
+
+	evt_test->assert_header();
+
+	/*=============================== ASSERT PARAMETERS  ===========================*/
+
+	/* Parameter 1: res (type: PT_INT64) */
+	evt_test->assert_numeric_param(1, (int64_t)0);
+
+	/* Parameter 2: pid (type: PT_PID) */
+	evt_test->assert_numeric_param(2, (int64_t)getpid());
+
+	/* Parameter 3: data (type: PT_BYTEBUF)*/
+	evt_test->assert_empty_param(3);
+
+	/*=============================== ASSERT PARAMETERS  ===========================*/
+
+	evt_test->assert_num_params_pushed(3);
+}
+
+TEST(SyscallExit, process_vm_writevX_success)
+{
+	auto evt_test = get_syscall_event_test(__NR_process_vm_writev, EXIT_EVENT);
+
+	evt_test->enable_capture();
+
+	/*=============================== TRIGGER SYSCALL ===========================*/
+
+	int pipe_fd[2];
+
+	ASSERT_GT(pipe(pipe_fd), -1);
+
+	pid_t child_pid = fork();
+
+	if(child_pid == 0)
+	{
+
+		char buf[10] = "QWERTYUIO";
+		struct iovec local[1];
+		local[0].iov_base = buf;
+		local[0].iov_len = sizeof(buf);
+		void* target;
+
+		close(pipe_fd[1]);
+
+		ssize_t read = syscall(__NR_read, pipe_fd[0], &target, sizeof(void*));
+		ASSERT_GT(read, 0);
+
+		read = syscall(__NR_read, pipe_fd[0], (void*)&child_pid, sizeof(pid_t));
+		ASSERT_GT(read, 0);
+
+		read = syscall(__NR_process_vm_writev, child_pid, local, 1, target, 1, 0);
+		assert_syscall_state(SYSCALL_SUCCESS, "process_vm_writev", read, NOT_EQUAL, 0);
+
+		close(pipe_fd[0]);
+
+		exit(EXIT_SUCCESS);
+	}
+	else
+	{
+
+		char buf[10];
+		struct iovec local[1];
+		local[0].iov_base = (void*)buf;
+		local[0].iov_len = sizeof(buf);
+		void* target = &local;
+
+		close(pipe_fd[0]);
+
+		ssize_t res = write(pipe_fd[1], &target, sizeof(void*));
+		ASSERT_GT(res, 0);
+
+		res = write(pipe_fd[1], (void*)&child_pid, sizeof(pid_t));
```

Comment:
Isn't this the PID of the child process? Since the child is also the one writing into remote memory, doesn't this mean that the child is writing in its own stack? I think that doing an `ASSERT_EQ((const char*) buf, "QWERTYUIO")` here below after having waited for the children to be exited should validate the theory. If I'm right, I think we should send the child process the value of `getpid()` instead of the one of `child_pid`. The child code is ok tho, as it just writes to the received pid's remote memory.

Reply:
You are right! When refactoring I probably loss a `child_pid = getpid();` somewhere :/